### PR TITLE
test: add admin and performance test and fix upgrade test

### PR DESCRIPTION
**Pull request by lumtis**
# Description

- Implement testing for CCTX performance test
- Fixes upgrade test for v15
- Add authority module in setup handler and fix migration
- Add some fixes in the existing localnet files
- Add light e2e test option to run upgrade tests

Closes: https://github.com/zeta-chain/node/issues/1668

### CODE REVIEW

**Comment by lumtis** [comment]
Some bugs in ZetaClient in upgrade tests currently happening, will wait for the ZetaClient refactorings before wrapping it up

**Comment by lumtis** [comment]
Performance tests need to be improved: https://github.com/zeta-chain/node/issues/1854

Already opening the PR so we can already ask DevOps to add the different tests in the CI

File: cmd/zetae2e/local/ethereum.go [hunk 1]
```diff
@@ -47,19 +49,20 @@ func ethereumTestRoutine(
...
-			e2etests.TestEtherWithdrawName,
-			e2etests.TestContextUpgradeName,
-			e2etests.TestEtherDepositAndCallName,
-			e2etests.TestDepositAndCallRefundName,
-			e2etests.TestEtherWithdrawRestrictedName,
+			//e2etests.TestEtherWithdrawName,
+			//e2etests.TestContextUpgradeName,
+			//e2etests.TestEtherDepositAndCallName,
+			//e2etests.TestDepositAndCallRefundName,
+			//e2etests.TestEtherWithdrawRestrictedName,
```
Comment (lumtis) [inline cmd/zetae2e/local/ethereum.go:64]:
```suggestion
```

File: changelog.md [hunk 2]
```diff
@@ -25,6 +25,7 @@
 * [1767](https://github.com/zeta-chain/node/pull/1767) - add unit tests for emissions module begin blocker
 * [1816](https://github.com/zeta-chain/node/pull/1816) - add args to e2e tests
 * [1791](https://github.com/zeta-chain/node/pull/1791) - add e2e tests for feature of restricted address
+* [1787](https://github.com/zeta-chain/node/pull/1787) - add unit tests for cross-chain evm hooks and e2e test failed withdraw to BTC legacy address
```
[the remark this answers is not in the document]
Comment (lumtis) [reply changelog.md:28]: Good point, don't hesitate to put a suggestion that can be merged directly on Github

File: cmd/zetae2e/config/clients.go [hunk 3]
```diff
@@ -80,7 +80,6 @@ func getBtcClient(rpcConf config.BitcoinRPC) (*rpcclient.Client, error) {
 		HTTPPostMode: rpcConf.HTTPPostMode,
 		DisableTLS:   rpcConf.DisableTLS,
 		Params:       param,
-		//Endpoint:     "/wallet/user",
```
[the remark this answers is not in the document]
Comment (lumtis) [reply cmd/zetae2e/config/clients.go:83]: Yes, it was for a test for Bitcoin in a previous PR

File: contrib/localnet/orchestrator/start-zetae2e.sh [hunk 4]
```diff
@@ -51,8 +51,19 @@ if [ "$OPTION" == "upgrade" ]; then
...
 
-  echo "running E2E command to setup the networks and populate the state..."
-  zetae2e "$ZETAE2E_CMD" --config-out deployed.yml
+  # Fetch the height of the upgrade, default is 200, if arg3 is passed, use that value
+  UPGRADE_HEIGHT=${3:-200}
+
+  # Run zetae2e, if the upgrade height is lower than 100, we use the setup-only flag
+  if [ "$UPGRADE_HEIGHT" -lt 100 ]; then
+    echo "running E2E command to setup the networks..."
+    zetae2e "$ZETAE2E_CMD" --setup-only --config-out deployed.yml
```
[the remark this answers is not in the document]
Comment (lumtis) [reply contrib/localnet/orchestrator/start-zetae2e.sh:60]: `if [ "$UPGRADE_HEIGHT" -lt 100 ]` might be an implicit condition but it is how you determine you run the light testing and require setup only. I tried to make it clear in the comment

File: contrib/localnet/orchestrator/start-zetae2e.sh [hunk 5]
```diff
@@ -63,37 +74,29 @@ if [ "$OPTION" == "upgrade" ]; then
...
   echo "running E2E command to test the network after upgrade..."
 
-  zetae2e "$ZETAE2E_CMD" --skip-setup --config deployed.yml
-
-  ZETAE2E_EXIT_CODE=$?
-  if [ $ZETAE2E_EXIT_CODE -eq 0 ]; then
-    echo "E2E passed after upgrade"
-    exit 0
+  # Run zetae2e again
+  # When the upgrade height is greater than 100 for upgrade test, the Bitcoin tests have been run once, therefore the Bitcoin wallet is already set up
```
[the remark this answers is not in the document]
Comment (lumtis) [reply contrib/localnet/orchestrator/start-zetae2e.sh:86]: It is part of the logic of the script

File: cmd/zetae2e/local/local.go [hunk 6]
```diff
@@ -236,14 +273,63 @@ func localE2ETest(cmd *cobra.Command, _ []string) {
...
+		eg.Go(erc20TestRoutine(conf, deployerRunner, verbose, erc20Tests...))
+		eg.Go(zetaTestRoutine(conf, deployerRunner, verbose, zetaTests...))
+		eg.Go(bitcoinTestRoutine(conf, deployerRunner, verbose, !skipBitcoinSetup, !light, bitcoinTests...))
+		eg.Go(ethereumTestRoutine(conf, deployerRunner, verbose, !light, ethereumTests...))
 	}
 	if testAdmin {
 		eg.Go(adminTestRoutine(conf, deployerRunner, verbose))
 	}
+	if testPerformance {
+		eg.Go(ethereumDepositPerformanceRoutine(conf, deployerRunner, verbose))
```
[the remark this answers is not in the document]
Comment (lumtis) [reply cmd/zetae2e/local/local.go:330]: Could point, let's use test name everywhere so all test names used in local testing are listed in `local.go`

**Review by kingpinXD — APPROVED** [review]

**Pull request merged**